iceoryx_posh/source/runtime/shared_memory_user.cpp

```diff
@@ -73,7 +75,23 @@ void SharedMemoryUser::openDataSegments(const uint64_t segmentId,
     {
         auto accessMode = segment.m_isWritable ? AccessMode::READ_WRITE : AccessMode::READ_ONLY;
         PosixSharedMemoryObjectBuilder()
-            .name(segment.m_sharedMemoryName)
+            .name([&segment] {
+                using ShmName_t = detail::PosixSharedMemory::Name_t;
+                ShmName_t shmName = iceoryxResourcePrefix(roudi::DEFAULT_UNIQUE_ROUDI_ID,
+                                                          "p"); // use an additional 'p' to prevent creating a payload
+                                                                // segment with the same name as the management segment
+                if (shmName.size() + segment.m_sharedMemoryName.size() > ShmName_t::capacity())
+                {
+                    IOX_LOG(FATAL,
+                            "The payload segment with the name '"
+                                << segment.m_sharedMemoryName.size()
+                                << "' would exceed the maximum allowed size when used with the '" << shmName
+                                << "' prefix!");
+                    IOX_PANIC("The shm name exceeds the max size with the 'iox1_#_p_' prefix");
```

Comment:
Here a prefix generating function would be nice as well to keep everything in sync when this is changed.

Reply:
Well, the prefix generating function is `iceoryxResourcePrefix` :sweat_smile: 

I would change it though to distinguish between iceoryx defined resources and user defined resources like proposed in https://github.com/eclipse-iceoryx/iceoryx/pull/2191#discussion_r1492724816